src/main/java/org/sonarlint/intellij/server/SonarLintHttpServer.kt

```diff
@@ -119,25 +121,29 @@ class ServerHandler : SimpleChannelInboundHandler<Any?>() {
     override fun channelRead0(ctx: ChannelHandlerContext, msg: Any?) {
         if (msg is HttpRequest) {
             origin = msg.headers()[HttpHeaderNames.ORIGIN]
-            response = RequestProcessor().processRequest(Request(msg.uri(), msg.method()))
+            response = RequestProcessor().processRequest(Request(msg.uri(), msg.method(), isTrustedOrigin(origin)))
         }
         if (msg is LastHttpContent) {
             ctx.writeAndFlush(createResponse(response, origin))
         }
     }
 
+    private fun isTrustedOrigin(origin: String?): Boolean {
+        return origin != null && (SonarLintUtils.isSonarCloudAlias(origin) || !Settings.getGlobalSettings().getConnectionsTo(origin).isEmpty())
```

Comment:
Spotted an issue while testing: the condition in `getConnectionsTo(String)` is not the right one for this need.

If the server is hosted under a directory - e.g http://some.host/dir - then the `Origin` will be http://some.host (without `/dir`). According to the `Origin` header specification, the correct check in our case would be to find a server connection whose `hostUrl` starts with the passed `Origin`.